src/ui/wxWidgets/optionspropsheet.cpp

```diff
@@ -1103,9 +1123,32 @@ void COptions::OnPWHistSaveClick( wxCommandEvent& /* evt */ )
 
 void COptions::OnPWHistApply( wxCommandEvent& evt )
 {
-  // XXX TBD - send this to someone who knows how to deal with it!
+  int applytoprotected = 1;
+
+  if (m_applytoprotectedCB->GetValue()) {
+    applytoprotected = -1;
+  }
+
+  if (m_pwhiststopRB->GetValue()) {
+    // Reset entries to HISTORY OFF
+    m_pwhistaction = 1 * applytoprotected;
+
+  } else if (m_pwhiststartRB->GetValue()) {
+    // Reset entries to HISTORY OFF
+    m_pwhistaction = 2 * applytoprotected;
+
+  } else if (m_pwhistsetmaxRB->GetValue()) {
+    // Don't reset history setting, but set history number
+    m_pwhistaction = 3 * applytoprotected;
+
+  } else if (m_pwhistclearRB->GetValue()) {
+    // Don't reset history setting, but clear all history items
+    m_pwhistaction = 4 * applytoprotected;
```

Comment:
Are these coefficient values (1,2,3,4) actually flags? Are they declared somewhere? If not, it would be nice to make them named constants.

Reply:
Not that I've seen.  I based this implementation on the Windows version, and this is how it handles it.  But I can go back and do that to both versions - that sounds like a good idea.

Reply:
- The 'apply' button is there as an attempt to keep close to the Windows implementation, good to keep it.
- There are a few places where numerical values are used instead of well-named constants. I wouldn't go through the entire code base looking to clean these up, but if you have your editor open on the relevant section, I'd appreciate the extra effort. Bonus points for doing this cross-platform :-)